### ISSUE TITLE
It is not possiable to changing user  avatar   

### ISSUE THREAD

**Issue by 375573088** [vector-im/element-android#4767]
### Steps to reproduce

It is not possiable to changing user  avatar
![image](https://user-images.githubusercontent.com/37624681/146704218-94e90dc1-edf2-4f2c-a6d4-502c296e7320.png)
![image](https://user-images.githubusercontent.com/37624681/146704251-147ca195-48df-4b90-b6e0-86781c8663fc.png)
cut and uploading photo
![image](https://user-images.githubusercontent.com/37624681/146704298-baec10e2-dd6d-4eca-9472-b9ab742d4b09.png)
![image](https://user-images.githubusercontent.com/37624681/146704499-8acff7f7-0003-42c9-beff-f668890f4a62.png)

### Outcome

final Exception：android.os.NetworkOnMainThreadException
![image](https://user-images.githubusercontent.com/37624681/146704435-bfed8aae-8b18-44db-b67e-164793b8c443.png)
![image](https://user-images.githubusercontent.com/37624681/146704499-8acff7f7-0003-42c9-beff-f668890f4a62.png)

### Your phone model

18208727060

### Operating system version

Android 31

### Homeserver

https://im.ynzzpt.cn/

### Will you send logs?

Yes

**Comment by YamatoRyou** [comment]
Same as me.
#4730

**Comment by 375573088** [comment]
Changes in Element v1.3.9 (2021-12-01)

**Comment by 375573088** [comment]
And i did't  have  made any modifications.
That is my   virtual  machine  device：
![image](https://user-images.githubusercontent.com/37624681/146772853-1212d058-8002-4580-ac12-15921f0e5c07.png)

**Comment by ouchadam** [comment]
interesting :thinking:  


- are you able to confirm if the exception is thrown by...

```
val response = fileUploader.uploadFromUri(...)
```

or 

```
setAvatarUrlTask.execute(...))
```

- and are you able to check if matrix.org works?

**Comment by 375573088** [comment]
This is try catch code，and execute the code  Exception：android.os.NetworkOnMainThreadException

```
  override suspend fun updateAvatar(userId: String, newAvatarUri: Uri, fileName: String) {
        val logger = FormattedJsonHttpLogger()
        withContext(coroutineDispatchers.main) {
           try{
               val response = fileUploader.uploadFromUri(newAvatarUri, fileName, MimeTypes.Jpeg)
               setAvatarUrlTask.execute(SetAvatarUrlTask.Params(userId = userId, newAvatarUrl = response.contentUri))
               userStore.updateAvatar(userId, response.contentUri)
           }
           catch (e :Exception){
               logger.log("updateAvatar------>"+ e.message)  // android.os.NetworkOnMainThreadException
           }
        }
    }
```

**Comment by ouchadam** [comment]
what happens if we only execute the file uploading? 

```kotlin
  override suspend fun updateAvatar(userId: String, newAvatarUri: Uri, fileName: String) {
        val logger = FormattedJsonHttpLogger()
        withContext(coroutineDispatchers.main) {
           try{
               fileUploader.uploadFromUri(newAvatarUri, fileName, MimeTypes.Jpeg)
               //setAvatarUrlTask.execute(SetAvatarUrlTask.Params(userId = userId, newAvatarUrl = response.contentUri))
               //userStore.updateAvatar(userId, response.contentUri)
           }
           catch (e :Exception){
               logger.log("updateAvatar------>"+ e.message)  // android.os.NetworkOnMainThreadException
           }
        }
    }
```

if we still get the `NetworkOnMainThreadException` then it proves the error is within `fileUploader.uploadFromUri`, if there is no error then we can infer that `setAvatarUrlTask.execute` is the problem

**Comment by 375573088** [comment]
The way just only execute this  code and exception：   
   
```
fileUploader.uploadFromUri(newAvatarUri, fileName, MimeTypes.Jpeg)
```

however it is  not step into below code：

```
setAvatarUrlTask.execute
```

**Comment by ouchadam** [comment]
I'm really confused by this issue, the only way I've been able to reproduce the `NetworkOnMainThreadException` in the `uploadFromUri` flow is by replacing `okHttpClient.newCall(request).awaitResponse()` with `okHttpClient.newCall(request).execute()

https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/content/FileUploader.kt#L144


@375573088 does changing the dispatcher to `.io` fix the exception?

```diff
- withContext(coroutineDispatchers.main) {
+ withContext(coroutineDispatchers.io) {
```

https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt#L71

**Comment by 375573088** [comment]
Very nice ，Thanks for you.   Use  ```coroutineDispatchers.io```  it is dit fixed the exception and setAvatar successed。
i guess `coroutineDispatchers.main`  it's can not dispatch my process into asynchronous process。
```
 withContext(coroutineDispatchers.io)
```
Now ， Wish you a merry Christmas in advance. 3Q

![image](https://user-images.githubusercontent.com/37624681/146855247-61052ca6-8bce-4481-be56-bdefbb92a6f2.png)
![image](https://user-images.githubusercontent.com/37624681/146855285-a09b7e24-40ce-4983-a653-bfa915a2e51b.png)
![image](https://user-images.githubusercontent.com/37624681/146855301-5f1a1a14-1606-406f-aab7-30cb3dea7f72.png)

**Comment by ouchadam** [comment]
:100: great to hear! I'll raise a PR with the change

thanks for your help with debugging

**Comment by 375573088** [comment]
> 💯 great to hear! I'll raise a PR with the change
> 
> thanks for your help with debugging

No problem，hahaha~

**Comment by bmarty** [comment]
Fix included in EA 1.3.13
